include/openssl/dh.h

```diff
@@ -185,21 +185,24 @@ DEPRECATEDIN_3_0(int DH_compute_key(unsigned char *key, const BIGNUM *pub_key,
                                     DH *dh))
 DEPRECATEDIN_3_0(int DH_compute_key_padded(unsigned char *key,
                                            const BIGNUM *pub_key, DH *dh))
-DECLARE_ASN1_ENCODE_FUNCTIONS_only(DH, DHparams)
-DECLARE_ASN1_ENCODE_FUNCTIONS_only(DH, DHxparams)
+
+DEPRECATEDIN_3_0(DH *d2i_DHparams(DH **a, const unsigned char **in, long len))
+DEPRECATEDIN_3_0(int i2d_DHparams(const DH *a, unsigned char **out))
+DEPRECATEDIN_3_0(DH *d2i_DHxparams(DH **a, const unsigned char **in, long len))
+DEPRECATEDIN_3_0(int i2d_DHxparams(const DH *a, unsigned char **out))
```

Comment:
Why not use `DECLARE_ASN1_ENCODE_FUNCTIONS_only_attr`?

Reply:
This was a carry over from an early iteration of this PR where I hadn't quite figured out how I need to do this properly. Now fixed.